web/src/main/java/org/perfrepo/web/adapter/exceptions/ConstraintViolationException.java

```diff
@@ -0,0 +1,19 @@
+package org.perfrepo.web.adapter.exceptions;
+
+/**
+ * This unchecked exception is thrown when service adapter could not perform action,
+ * because it would cause constraint violation.
+ *
+ * @author Jiri Grunwald (grunwjir@gmail.com)
+ */
+public class ConstraintViolationException extends AdapterException {
```

Comment:
I don't understand it's purpose. Shouldn't ValidationException handle improper properties? In case that the validation passes, the operation should also pass. If it doesn't, it's an error in the backend not cause by user, hence the application should just signal something like "Internal server error" wrapped into AdapterException. Or what do you think?